### PR TITLE
Update minimum NumPy version to v1.26.4.

### PR DESCRIPTION
**Pull request by hawkinsp**
Remove code to handle NumPy 2.0 prereleases.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,9 +17,7 @@ keywords = []
 # pip dependencies of the project
 dependencies = [
     # Ensure numpy release supports Python version.
-    "numpy>1.20",
-    "numpy>=1.21.2; python_version>='3.10'",
-    "numpy>=1.23.3; python_version>='3.11'",
+    "numpy>1.24",
```
Comment (jakevdp) [inline pyproject.toml:20]: Should this be `numpy>=1.24`?

**Review by jakevdp — COMMENTED** [review]
Is there any cost to us in continuing to support older NumPy versions? If not I'd lean towards keeping support for older versions for now.

**Comment by hawkinsp** [comment]
> Is there any cost to us in continuing to support older NumPy versions? If not I'd lean towards keeping support for older versions for now.

Nothing other than following NEP 29 (and its successor).

**Comment by hawkinsp** [comment]
Closing since we decided we'd not upgrade so aggressively.

**Pull request closed without merging**